attributes/default.rb

```diff
@@ -11,13 +11,18 @@
 default[:hadoop][:tmp_dir]                 = "#{node[:hadoop][:home]}/tmp"
 default[:hadoop][:conf_dir]                = "#{node[:hadoop][:home]}/etc/hadoop"
 default[:hadoop][:sbin_dir]                = "#{node[:hadoop][:home]}/sbin"
-default[:hadoop][:bin_dir]                = "#{node[:hadoop][:home]}/bin"
+default[:hadoop][:bin_dir]                 = "#{node[:hadoop][:home]}/bin"
 default[:hadoop][:data_dir]                = "/var/data/hadoop"
 default[:hadoop][:dn][:data_dir]           = "#{node[:hadoop][:data_dir]}/hdfs/dn"
 default[:hadoop][:nn][:name_dir]           = "#{node[:hadoop][:data_dir]}/hdfs/nn"
 
-default[:hadoop][:download_url]            = "https://archive.apache.org/dist/hadoop/core/hadoop-#{node[:hadoop][:version]}/hadoop-#{node[:hadoop][:version]}.tar.gz"
-#default[:hadoop][:download_url]            = "#{node[:download_url]}/hadoop-#{node[:hadoop][:version]}.tar.gz"
+default[:hadoop][:download_url]            = "ftp://ftp.fu-berlin.de/unix/www/apache/hadoop/common/hadoop-#{node[:hadoop][:version]}/hadoop-#{node[:hadoop][:version]}.tar.gz"
+version = "#{node[:hadoop][:version]}"
+version.gsub!('.', '')
```

Comment:
That's correct. Actually though, it's kind of a hack. Let me change the
behavior such that there is a primary and a secondary URL and the primary
is always checked first. If the binaries can't be obtained from the primary
URL, check the secondary instead. I think that makes more sense. Do you
agree?

I'll create a new pull request as soon as I'm done with it.

On Tue, Nov 10, 2015 at 12:52 PM, Jim Dowling <notifications@github.com>
wrote:

> In attributes/default.rb
> <https://github.com/hopshadoop/apache-hadoop-chef/pull/6#discussion_r44397613>
> :
>
> >  default[:hadoop][:data_dir]                = "/var/data/hadoop"
> >  default[:hadoop][:dn][:data_dir]           = "#{node[:hadoop][:data_dir]}/hdfs/dn"
> >  default[:hadoop][:nn][:name_dir]           = "#{node[:hadoop][:data_dir]}/hdfs/nn"
> >
> > -default[:hadoop][:download_url]            = "https://archive.apache.org/dist/hadoop/core/hadoop-#{node[:hadoop][:version]}/hadoop-#{node[:hadoop][:version]}.tar.gz"
> > -#default[:hadoop][:download_url]            = "#{node[:download_url]}/hadoop-#{node[:hadoop][:version]}.tar.gz"
> > +default[:hadoop][:download_url]            = "ftp://ftp.fu-berlin.de/unix/www/apache/hadoop/common/hadoop-#{node[:hadoop][:version]}/hadoop-#{node[:hadoop][:version]}.tar.gz"
> > +version = "#{node[:hadoop][:version]}"
> > +version.gsub!('.', '')
>
> This is taking recent versions for fu-berlin, and old versions from hadoop
> archive. Is that correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hopshadoop/apache-hadoop-chef/pull/6/files#r44397613>.
>